configure.py

```diff
@@ -68,6 +69,9 @@
     "Flags that allows you to control whether you want to build against "
     "built-in dependencies or its shared representations. If necessary, "
     "provide multiple libraries with comma.")
+static_optgroup = parser.add_argument_group("Static libraries",
```

Comment:
In this case for z/OS we always build against the additional library right?  Is the issue that it won't be in a know location within the OS?

Reply:
In the upstream V8, zoslib is a dependency that will be downloaded (by `gclient sync`) and placed in `v8/third_party/zoslib`. There, I also build zoslib (auto, from `v8/BUILD.gn`) as a shared library (and all V8 executables link with its sidedeck `libzoslib.x`). Should we build Node.js based on zoslib being in `deps/v8/third_party/zoslib` and also access it as shared library instead of static as currently done?

Reply:
One consideration is that we have some C lib function overrides that will only work when zoslib is linked statically: https://github.ibm.com/open-z/zoslib/blob/master/src/zos.cc#L678

But there is a way to workaround this.

Reply:
> Should we build Node.js based on zoslib being in deps/v8/third_party/zoslib and also access it as shared library instead 
of static as currently done?

My answer to the first part would be yes, but for the second part it is more consistent to link in dependencies statically so that you don't need to install any other libraries for the node.js executable to work.  If we linked dynamically there would then be the question of how you install the shared library into the OS and will Node.js find it?

Reply:
I see @IgorTodorovskiIBM comment also points to linking statically. 

I might revise my comment on the first part.  We don't build Node.js with gclient sync.

I think we should be linking statically, but the question would be if it should be considered an OS dependency. ie you need to install it into your z/OS env to be able to build, and then assume it has been installed into a standard location?

Reply:
For building in the community we would then need to makes sure that the community ansiblescripts installed into the right place.

Reply:
> ie you need to install it into your z/OS env to be able to build, and then assume it has been installed into a standard location?

How about that standard location be `deps/v8/third_party/zoslib` (even if for Node.js we'd manually clone it into that location)? In Node.js, only `libv8_libbase.a` contains `libzoslib.a`, and `node` binary (no longer) links with `libzoslib.a` since it (through `libv8_libbase.a`) is already in `libnode.so`. 

> I think we should be linking statically

fyi - we currently link Node.js statically, but also ship `libzoslib.so` under `install/lib`, and updated the user's `.nodeenv` to set `LIBPATH`; that's because in `deps/v8/tools/testrunner/local/utils.py`, `IsS390SimdSupported()` uses an API from zoslib, so it loads `libzoslib.so` to call that API.

Reply:
zlib have a similar problem that both v8 and nodejs requires it, the current way is that both v8 and nodejs have and use its own zlib, but it could be configured that v8 and node.js both use the node.js's zlib. Perhaps we can put a mirrored zoslib under `deps` which nodejs's v8 will also use, but since zoslib only required by z/OS, I don't know would it be a welcome change to the community.

Reply:
I may not understand libzoslib.so well enough and related issues but the model I had in mind was 

1) install os lib, it goes into standard location for os like /usr/lib
2) Link against library in that standard location.  

I don't think we want the complication of suggesting a new dep under the node.js ../deps tree if at all possible.

Maybe we can talk it through in our meeting on Monday.